Added YouTube and Spotify support

dillonupgradeit opened the pull request:
Co-Authored-By: cmgriffing <cmgriffing@gmail.com>

Tried to label most of the code with comments, should be pretty easy to go through. I followed your lead. Remember I'm new to JS so any suggestions are completely welcome!

dillonupgradeit committed:
Added YouTube and Spotify support

Co-Authored-By: cmgriffing <cmgriffing@gmail.com>

cmgriffing commented on background.js:
```diff
@@ -16,47 +21,146 @@ setInterval(() => {
+              body.origin = scrapedSong[0].origin;
+              body.domain = scrapedSong[0].domain;
+              if(currentSong == scrapedSong[0].songName && currentSongTime == scrapedSong[0].currentTime){
+              
```
A comment here explaining why we do nothing would be cool.

Couldn't the `if` logic just be inverted though?

cmgriffing commented on background.js:
```diff
@@ -16,47 +21,146 @@ setInterval(() => {
+          scrapedSong => {
+            console.log('scrapedSong', scrapedSong);
+            if (scrapedSong[0]) {
+              body.songName = scrapedSong[0].currentSongApprox!==undefined?scrapedSong[0].currentSongApprox:scrapedSong[0].songName; //USE APPROX UNLESS EMPTY
```
I think this is a bit verbose for a ternary and it wuld need spacing anyway.

How about an `if` instead?

cmgriffing commented on background.js:
```diff
@@ -16,47 +21,146 @@ setInterval(() => {
     .catch(error => {
       console.log('something has gone terribly wrong', error);
+      var noBody = {};
+      noBody.origin = "NONE";
```
Couldn't this just be done in the declaration?

```
var noBody = { origin: "NONE" };
```

It could also probably just be used directly as an argument without the need for a variable.

cmgriffing commented on background.js:
```diff
@@ -105,6 +220,7 @@ const postData = (url = ``, data = {}) => {
     credentials: 'same-origin', // include, same-origin, *omit
     headers: {
       'Content-Type': 'application/json; charset=utf-8',
+      //'Content-Type' : 'application/x-www-form-urlencoded; charset=UTF-8'
```
Do we need this line here?

cmgriffing commented on docker-compose.yml:
```diff
@@ -5,4 +5,4 @@ services:
       dockerfile: server.dockerfile
     ports:
-      - "4242:4242"
+      - "8293:8293"
```
Any reason why this was changed? doesn't this change the url we post to from the frontend?

dillonupgradeit committed:
Bandcamp custom Domains, album artwork, and frontend interface

Get song info from BandCamp sites with a Custom Domain
Get Image URL from Youtube, Spotify, and BandCamp
YouTube get channelName, and more reliable artist name and songName
Popup: display current song info/artwork and ON/OFF switch.
Options Page: customize line sent to BOT (including if statements), ability to disable specific platforms, and customize bot server/port info

dillonupgradeit committed:
Push Upstream to Origin Master

Push Upstream Master to Origin Master

dillonupgradeit committed:
Add Page Types to Bandcap

Additional page types include: singles, fan/collelction pages, featured pages, bandcamp weekly pages, and homepage discover.

dillonupgradeit committed:
Refactor Version Number

Refactor version number.

dillonupgradeit committed:
Refactor Cleanup

Refactor comments and spacing.

dillonupgradeit committed:
Merge from Master

Merge from master and resolve conflicts.

dillonupgradeit committed:
Remove use of var from get-song-info.js

Remove use of var from get-song-info.js

dillonupgradeit committed:
Wrap get-song-info.js in function

Wrap get-song-info.js in function to avoid redeclared constants error.

dillonupgradeit committed:
Update albumTitle node on single/track pages

Update albumTitle node on single/track pages with backup itemprop nodes.

dillonupgradeit committed:
Update bandcamp weekly to include playlist name and url

dillonupgradeit committed:
Update get-song-info.js discover to utilize discover detail player

dillonupgradeit committed:
Fix issue with homepage and bcweekly-info-inner panel

Bandcamp hompage also displayed bcweekly-info-inner, so you cannot rely on this to determine if Bandcamp Weekly Show page.

dillonupgradeit committed:
Add artistURL to tag, fan, and collection pages

dillonupgradeit committed:
Fix issue with album URL on hompage discover

dillonupgradeit committed:
Add currentTime and durationTime

Add currentTime and durationTime to album + single/track pages, tags + fans + collections pages, and bandcamp featured page

dillonupgradeit committed:
Add commented descriptions to get-song-name.js

dillonupgradeit committed:
Fix currentTime and durationTime on Bandcamp weekly page

dillonupgradeit committed:
Merge branch 'Add-Page-Types-to-Bandcamp'

dillonupgradeit committed:
Update Version Number

dillonupgradeit committed:
Update README.md

dillonupgradeit committed:
Update README.md

Update README.md

Update README.md

Update README.md

dillonupgradeit committed:
Update README.md

dillonupgradeit committed:
Update README.md

dillonupgradeit committed:
Update README.md

dillonupgradeit committed:
Correct Version Number

dillonupgradeit committed:
Correct Version Number

dillonupgradeit committed:
Add Changelog to README.md

dillonupgradeit committed:
Update README.md

dillonupgradeit committed:
Merge branch 'Add-Page-Types-to-Bandcamp'

dillonupgradeit committed:
Update README.md

dillonupgradeit committed:
Update get-song-info.js filename to clarify

dillonupgradeit committed:
Update get-song-info.js filename to clarify

dillonupgradeit committed:
Merge branch 'Add-Page-Types-to-Bandcamp'

dillonupgradeit committed:
Update README.md

dillonupgradeit committed:
Update README.md

dillonupgradeit committed:
Update README.md

dillonupgradeit committed:
Merge branch 'Add-Page-Types-to-Bandcamp' of https://github.com/dillonupgradeit/now-playing into Add-Page-Types-to-Bandcamp

dillonupgradeit committed:
Merge branch 'Add-Page-Types-to-Bandcamp'

dillonupgradeit committed:
Add initial spotify router

dillonupgradeit committed:
Update get-spotify-info.js

dillonupgradeit committed:
Update version number

dillonupgradeit committed:
Update README.md
